### ISSUE TITLE
Subquery IS [ NOT ] NULL predicate should avoid RVE emulation for degree = 1

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11011]
A recently added integration test fails on Derby:

```java
TA a = T_A.as("a");

List<Integer> r1 =
create().select(a.A)
        .from(a)
        .where(select(a.tA().B1).isNotNull())
        .orderBy(a.A)
        .fetch(a.A);
```

The resulting query is this:

```sql
select a.A
from (
  T_A as a
    left outer join T_A as alias_53273784
      on a.PARENT_A = alias_53273784.A
  )
where 1 = (
  select count(*)
  from (
    select alias_53273784.B1 as v0
    from SYSIBM.SYSDUMMY1
  ) as t
  where t.v0 is not null
)
order by a.A
```

Derby (and a few other dialects) cannot nest queries correlated subqueries 2 levels deep. As such, we're getting this error:

> Column 'ALIAS_53273784.B1' is either not in any table in the FROM list or appears within a join specification and is outside the scope of the join specification or appears in a HAVING clause and is not in the GROUP BY list. If this is a CREATE or ALTER TABLE  statement then 'ALIAS_53273784.B1' is not a column in the target table.

We can certainly have a more optimal emulation of this query than the recycled `RowIsNull` query, given that in this particular case, we don't have any row value expressions.

**Comment by lukaseder** [comment]
Commit here: https://github.com/jOOQ/jOOQ/commit/00e6517dc6ee92ad6f4834fe7968c92d9f1863ad

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.4 (#11012)
